extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -50,64 +59,90 @@ export const HistoryItem = ({
     .split(" ")
     .slice(1, 3)
     .join(" ");
-  const operationAssetCode = assetCode || "XLM";
+  const srcAssetCode = sourceAssetCode || "XLM";
+  const destAssetCode = assetCode || "XLM";
 
   let isRecipient = false;
   let paymentDifference = "";
+  let rowText = "";
+  let dateText = "";
   let IconComponent = <Icon.Shuffle className="HistoryItem__icon--default" />;
   let PaymentComponent = null as React.ReactElement | null;
 
-  if (isPayment) {
+  let transactionDetailProps: TransactionDetailProps = {
```

Comment:
did a bit of refactoring here. Because now there's 3 states (swap, payment, or neither), I think cleaner to set each var in an if-elseif-else rather than a ternary for each one when used